### PR TITLE
[AIR] Add `TODO` to `from_checkpoint`

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

`from_checkpoint` is a hack that should be removed. I've added a note to the documentation to provide context.

## Related issue number

See #29058 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/checkpoint.py [hunk 1]
```diff
@@ -454,7 +454,8 @@ def from_directory(cls, path: str) -> "Checkpoint":
 
         return checkpoint
 
-    @classmethod
+    @classmethod  
+    # TODO: Deprecate `from_checkpoint`. For context, see #29058.
```
Comment (matthewdeng) [inline python/ray/air/checkpoint.py:458]: If this is something that's already aligned on and there is an alternative and better solution, should we go ahead and mark this as `@Deprecated`?

Comment (bveeramani) [reply python/ray/air/checkpoint.py:457]: Followed up on this offline -- we can't deprecate this until #29857 is fixed

Comment (matthewdeng) [reply python/ray/air/checkpoint.py:457]: Can you update https://github.com/ray-project/ray/issues/29058 to reflect that? It currently says:
> from_checkpoint is a hack. Now that we've addressed the underlying issue, we should remove it.

and

> So, Checkpoint.from_checkpoint is now obsolete.

Comment (bveeramani) [reply python/ray/air/checkpoint.py:457]: Thanks. Update the issue.

**Review by amogkam — APPROVED** [review]

**Pull request merged**